aikau/src/test/resources/testApp/js/aikau/testing/mockservices/UploadMockXhr.js

```diff
@@ -43,23 +45,47 @@ define(["dojo/_base/declare",
        * @instance
        */
       setupServer: function alfresco_testing_MockXhr__setupServer() {
-         try
-         {
-            var responseCode = this.responseCode;
-            this.server.respondWith(function(request) {
-               request.uploadProgress({
-                  total: 100,
-                  loaded: 20
-               });
-               request.respond(responseCode, 
-                              {"Content-Type":"application/json;charset=UTF-8"},
-                               "{\"nodeRef\":\"bob\",\"fileName\":\"moomin\"}");
-            });
-         }
-         catch(e)
-         {
+         try {
+            this.server.respondWith(lang.hitch(this, function(request) {
+
+               // Setup headers/body for response                  
+               var headers = {
+                     "Content-Type": "application/json;charset=UTF-8"
+                  },
+                  body = "{\"nodeRef\":\"bob\",\"fileName\":\"moomin\"}";
+
+               // If response code isn't 200, respond immediately
+               if (this.responseCode !== 200) {
+                  request.respond(this.responseCode, headers, body);
+                  return;
+               }
+
+               // Randomly increment the progress every few moments until it's at 100 percent
+               var i;
+               request.readyState = 4; // Set the response to pretend to be completed, so Sinon doesn't send its own responses and close it before we finish
+               for (i = 1; i < 100; i += Math.round(Math.random() * 10)) {
+                  this.sendProgress(request, i, i * 20);
+               }
+
+               // After progress has finished, send the final response
+               setTimeout(function() {
+                  request.readyState = 1; // Sinon won't send response unless it thinks the response is clean and only just opened
+                  request.respond(this.responseCode, headers, body);
+               }, i * 20);
+
+            }));
+         } catch (e) {
             this.alfLog("error", "The following error occurred setting up the mock server", e);
          }
+      },
+
+      sendProgress: function(request, percent, delay) {
```

Comment:
I know it won't appear in the JSDoc output (at least not yet, although we did move MockXhr)... it would be nice to have the JSDoc statement here for consistency.